src/content/en/2022/javascript.md

```diff
@@ -16,7 +16,864 @@ featured_stat_2: TODO
+
+While first-party JavaScript may be easier to optimize, third-party JavaScript can itself be a significant source of performance problems, as third-party vendors may not prioritize the optimization of their JavaScript resources over adding new features to serve additional business functions for their clients. Additionally, UX researchers, marketers, and other non-technical personnel may be hesitant to give up functionality or sources of revenue that these scripts provide.
+
+In this section, we'll analyze the breakdown of first-party and third-party code, and comment on the current state of how websites today are divvying up how they load JavaScript—and where from.
+
+#### Requests
+
+{{ figure_markup(
+    image="requests-party.png",
+    caption="Distribution of first- versus third-party JavaScript requests by host.",
+    description="TODO",
+    chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTok8UGqYaA58uKaASB2pfM2jlmQu6g3kwHxB6Lb8L7dbccyDtQht823YhchdXRG8SZpB3asRayJI97/pubchart?oid=2001603308&format=interactive",
+    sheets_gid="755344459",
+    sql_file="requests_by_3p.sql"
+  )
+}}
+
+Here, we see a sobering picture. Regardless of the percentile, it seems that all observed hosts are serving an equivalent amount of first and third-party scripts. The median host serves 10 of each type, the 75th percentile serves 20 of each type, and the 90th percentile host serves _34 third-party scripts_.
+
+This is a problematic and worrying trend. Third-party scripts are responsible for all sorts of damage when it comes to performance. Third-party scripts may do a number of things, such as running expensive timers that orchestrate multitudes of tasks, attach their own event listeners that add extra work which can delay interactivity, and some video and social media third-parties ship exorbitant amounts of scripts to power the services they provide.
+
+Steps for mitigating third-party scripts is often more of a cultural affair than an engineering one. If you're shipping excessive third-party scripts, conduct an audit of each script, what they do, and profile their activity to find out what performance problems they're incurring.
+
+If you're doing considerable UX research, consider collecting your own field data (if the origin sends a proper [`Timing-Allow-Origin`](https://developer.mozilla.org/docs/Web/HTTP/Headers/Timing-Allow-Origin) header) to make informed decisions to avoid the performance problems that some third-party scripts can cause. For every third-party script you add, you're not just incurring loading costs, but also costs during runtime where responsiveness to user input is crucial.
+
+#### Bytes
+
+So we know that hosts are shipping a lot of third-party scripts, but what's the byte cost of first- versus third-party scripts?
```

Comment:
```suggestion
So we know that hosts are shipping a lot of third-party scripts, but what's the byte cost of first versus third-party scripts?
```

Reply:
I think the dangling hyphen is correct?

Reply:
Seems it's debated: https://onlinewritingtraining.com.au/hanging-hyphens/

But after checking, it's in our style guide: https://developers.google.com/style/hyphens#suspended-hyphens. Another thing I disagree with. Wonder if it's a UK versus US thing?

Anyway resolving this as you're right by our style guide.